src/List.js

```diff
@@ -1,119 +1,154 @@
 
 function List(props) {
-  const [data, setData] = useState(props.listItems);
-  const [idCounter, setIdCounter] = useState(data.length);
-  const [isChecked, setIsChecked] = useState([]);
+  const [numChecked, setNumChecked] = useState(0);
+  const [totalTasks, setTotalTasks] = useState(0);
+  const [editingText, setEditingText] = useState('');
   const [isEditingId, setIsEditingId] = useState(null);
-  const [editingText, setEditingText] = useState("");
   const [showAlert, setShowAlert] = useState(false);
   const [showingAllTasks, setShowingAllTasks] = useState(true);
-
-  function handleAdd(e) {
-    console.log("adding");
-    setData([...data, {
-      id: idCounter,
-      task: ""
-    }]);
-    setIsEditingId(idCounter);
-    setEditingText("");
-    setIdCounter(idCounter + 1);
+  const [newestItem, setNewestItem] = useState(null);
+
+  useEffect(() => {
+    const allSnap = props.collection.get();
+    allSnap.then(snap => setTotalTasks(snap.size));
+    const checkedSnap = props.collection.where('checked', '==', true).get();
+    checkedSnap.then(snap => setNumChecked(snap.size));
+  }, [props.collection]);
+
+  function handleAdd() {
+    const newId = generateUniqueID();
+    const newTask = {
+      id: newId,
+      created: firebase.database.ServerValue.TIMESTAMP,
+      task: '',
+      priority: '3',
+      checked: false
+    };
+    props.collection.doc(newId).set(newTask);
+    setNewestItem(newId);
   }
 
   function handleEditClick(e) {
-    setIsEditingId(e.target.id);
+    console.log('handleeditclick');
     setEditingText(e.target.value);
+    setIsEditingId(e.target.id);
+    setNewestItem(null);
   }
 
   function handleEditChange(e) {
+    console.log('handleeditchange');
     setEditingText(e.target.value);
-    let newData = data.map(item =>
-      parseInt(item.id) === parseInt(isEditingId)
-      ? { ...item, task: editingText }
-      : item);
-    setData(newData);
+    props.collection.doc(e.target.id).set({ task: editingText }, { merge: true });
   }
 
   function handleEditComplete(e) {
-    let newData = data.map((item) =>
-      parseInt(item.id) === parseInt(isEditingId)
-        ? { ...item, task: editingText }
-        : item
-    );
-    setData(newData);
-    setEditingText("");
+    props.collection.doc(e.target.id).set({ task: editingText }, { merge: true });
     setIsEditingId(null);
   }
 
-  function handleEditEnter (e) {
-    if (e.key === "Enter") {
+  function handleEditEnter(e) {
+    if (e.key === 'Enter') {
+      handleEditComplete({ target: { id: e.target.id } });
       e.target.blur();
-      handleEditComplete(e);
     }
   }
 
-  function handleIsCheckedChange(e) {
-    let newId = parseInt(e.target.id);
-    if (isChecked.includes(newId)) {
-      setIsChecked(isChecked.filter((id) => id !== newId));
-    } else {
-      setIsChecked([...isChecked, newId]);
-    }
+  async function handleIsCheckedChange(e) {
+    const docRef = props.collection.doc(e.target.id);
```

Comment:
You already know whether or not this item is now checked: it's in e.target.checked. 

No need for the get or the checked.

Also, no need for the await.